### PR TITLE
Add ARP santizer

### PR DESCRIPTION
**Pull request by parijke**
Added a local function to remove al irrelevant attributes from the ARP returned from the API calls. This way, the API can "grow" without erroring the Arp class

### CODE REVIEW

File: src/OpenConext/Profile/Value/Arp.php [hunk 1]
```diff
@@ -40,11 +40,12 @@ public static function createWith(
...
     ): self {
+
+        $arp = self::sanitizeARPData($arp);
+
         // Input validation
         foreach ($arp as $attributeInformation) {
-            if (!is_array($attributeInformation)) {
-                throw new InvalidArpDataException('The attribute information in the arp should be an array.');
-            }
+
```
Comment (parijke) [inline src/OpenConext/Profile/Value/Arp.php:48]: This if is moved inot the sanitizer

File: src/OpenConext/Profile/Value/Arp.php [hunk 2]
```diff
@@ -186,4 +187,40 @@ public function getMotivationFor(
         }
         return '';
     }
+
+    /**
+     * Accordingly to https://www.pivotaltracker.com/n/projects/1453004/stories/187607790
+     * the ARP data should only contain the following keys: value, source, motivation.
+     * This function will filter out all other keys, which are to be ignored.
+     */
+    private static function sanitizeARPData(array $arp): array
```
Comment (MKodde) [inline src/OpenConext/Profile/Value/Arp.php]: Nitpick: sanitizeArpData. Abbreviations are considered normal words.

File: src/OpenConext/Profile/Tests/fixture/arp-response.json [hunk 3]
```diff
@@ -66,5 +66,12 @@
     {
       "value": "*"
     }
+  ],
+  "urn:mace:dir:attribute-def:eduPersonAffiliation": [
```
Comment (MKodde) [inline src/OpenConext/Profile/Tests/fixture/arp-response.json:70]: This attribute is already in the fixture, may I suggest you pick a different attr name here?

Also, since we allow the `motivation` field too, that (sh|c)ould be covered in this test too?

File: src/OpenConext/Profile/Value/Arp.php (hunk 1 above)
Comment (MKodde) [reply src/OpenConext/Profile/Value/Arp.php]: In the end, you did not move it to the sanitizer right?

File: src/OpenConext/Profile/Value/Arp.php [hunk 4]
```diff
@@ -186,4 +187,40 @@ public function getMotivationFor(
...
+        $sanitizedArp = [];
+        foreach ($arp as $attributeName => $attributeInformation) {
+            if (!is_array($attributeInformation)) {
+                throw new InvalidArpDataException('The attribute information in the arp should be an array.');
+            }
+
+            $sanitizedAttributeInformation = [];
+            foreach ($attributeInformation as $attributeInformationEntry) {
+                if (!is_array($attributeInformationEntry)) {
+                    continue;
```
Comment (MKodde) [inline src/OpenConext/Profile/Value/Arp.php:209]: Is this actually occurring? This feels like a corner case where something bigger is wrong.  Also, given you added this bit of behavior, its only fair to also cover it in test.

File: src/OpenConext/Profile/Value/Arp.php [hunk 5]
```diff
@@ -186,4 +187,40 @@ public function getMotivationFor(
...
+                }
+
+                $sanitizedAttributeInformation[] = array_filter(
+                    $attributeInformationEntry,
+                    function ($key) use ($validKeys) {
+                        return in_array($key, $validKeys);
+                    },
+                    ARRAY_FILTER_USE_KEY
+                );
```
Comment (MKodde) [inline src/OpenConext/Profile/Value/Arp.php]: :ok_hand:

**Review by MKodde — CHANGES_REQUESTED** [review]
Nice work @parijke. I feel some i's need some dotting.

File: src/OpenConext/Profile/Value/Arp.php (hunk 1 above)
Comment (parijke) [reply src/OpenConext/Profile/Value/Arp.php]: @MKodde yes, 206-208

File: src/OpenConext/Profile/Tests/fixture/arp-response.json (hunk 3 above)
Comment (parijke) [reply src/OpenConext/Profile/Tests/fixture/arp-response.json:70]: @MKodde As I have no clue what attributes really are, can you give me a suggestion? Or can it be anything?

File: src/OpenConext/Profile/Value/Arp.php (hunk 4 above)
Comment (parijke) [reply src/OpenConext/Profile/Value/Arp.php:213]: It is, as it was in the create before.... In the test (see ArpTest line 102) it has thios edge...

File: src/OpenConext/Profile/Tests/fixture/arp-response.json (hunk 3 above)
Comment (parijke) [reply src/OpenConext/Profile/Tests/fixture/arp-response.json:70]: And sure, I can add motivation as well (as this wasn't subject of this use case, I did not think about this)

**Review by parijke — COMMENTED** [review]
@MKodde responed on your spot-on review!

File: src/OpenConext/Profile/Value/Arp.php (hunk 1 above)
Comment (MKodde) [reply src/OpenConext/Profile/Value/Arp.php]: Thanks for pointing it out. :+1:

File: src/OpenConext/Profile/Value/Arp.php (hunk 2 above)
Comment (MKodde) [reply src/OpenConext/Profile/Value/Arp.php]: The calling part must also be updated now ;) 

TIL: calling methods is not case sensitive :thinking:

File: src/OpenConext/Profile/Tests/fixture/arp-response.json (hunk 3 above)
Comment (MKodde) [reply src/OpenConext/Profile/Tests/fixture/arp-response.json]: The attributes are released by the IdP and they contain the data that describe the user. For example there is a mail attribute, a username, commonname, lastname.. You name e'm, there is an attribute for that. And if it does not exist, we can add them to the attribute definition.

I do not know all attributes by memory. I suggest you look in the attribute dictionary file. And pick one from there?

@see: vendor/surfnet/stepup-saml-bundle/src/Resources/config/saml_attributes.yaml

**Review by MKodde — COMMENTED** [review]
Almost there. Two minor todo's remain

File: src/OpenConext/Profile/Tests/fixture/arp-response.json (hunk 3 above)
Comment (parijke) [reply src/OpenConext/Profile/Tests/fixture/arp-response.json]: I added "motivation" to one of the testcases

File: src/OpenConext/Profile/Value/Arp.php (hunk 2 above)
Comment (parijke) [reply src/OpenConext/Profile/Value/Arp.php]: Apparantly (didn't knew that either). Will fix though

**Review by thijskh — APPROVED** [review]
I've tested this and it indeed solves the problem. Noticed that the newly added test is still failing though.

**Review by MKodde — APPROVED** [review]
